starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/DisplayActivity.java

```diff
@@ -0,0 +1,34 @@
+package ly.generalassemb.drewmahrt.shoppinglistdetailview;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+import org.w3c.dom.Text;
+
+public class DisplayActivity extends AppCompatActivity {
+    public static final String ITEM_ID_KEY = "item_id_key";
+    TextView nameView, descView, priceView, typeView;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_display);
+
+        nameView = (TextView)findViewById(R.id.name_view);
+        descView =(TextView)findViewById(R.id.descript_view);
+        priceView = (TextView)findViewById(R.id.price_view);
+        typeView = (TextView)findViewById(R.id.type_view);
+
+
+        int idKey = getIntent().getIntExtra(ITEM_ID_KEY, -1);
+
+        if (idKey == -1){
+            finish();
```

Comment:
Instead of automatically kicking the user back to the previous screen if there is an error, consider showing them an error message first so they know something went wrong.